modules/administration-guide/partials/proc_setting-up-the-github-oauth-app.adoc

```diff
@@ -0,0 +1,52 @@
+:_content-type: PROCEDURE
+:description: Setting up the GitHub OAuth App
+:keywords: github, github-oauth-app
+:navtitle: Setting up the GitHub OAuth App
+// :page-aliases:
+
+[id="setting-up-the-github-oauth-app_{context}"]
+= Setting up the GitHub OAuth App
+
+Set up a GitHub OAuth App using OAuth 2.0.
+
+.Prerequisites
+
+* You are logged in to GitHub.
+* link:https://www.gnu.org/software/coreutils/base64[`base64`] is installed in the operating system you are using.
+
+.Procedure
+
+. Go to link:https://github.com/settings/applications/new[].
+//Long version: Click your GitHub avatar menu:Settings[Developer settings > OAuth Apps > Register a new application]. max-cx
+
+. Enter *{prod-short}* as the *Application name*.
+
+. Enter `pass:c,a,q[{prod-url}]/` as the *Homepage URL*.
+
+. Enter `pass:c,a,q[{prod-url}]/api/oauth/callback` as the *Authorization callback URL*.
+
+. Click *Register application*.
+
+. Click *Generate new client secret*.
+
+. Copy the *GitHub OAuth Client ID* and encode it to Base64:
++
+[subs="+quotes,+attributes,+macros"]
+----
+$ echo -n '__<github_oauth_client_id>__' | base64
```

Comment:
The implementation of the `base64` tool is inconsistent across operating systems. It may add an undesired newline. Therefore it is better to document the conversion to the base64 format using the `openssl` tool, that has consistent behavior.

`$ openssl base64 -e -A `

See: https://www.eclipse.org/che/docs/che-7/end-user-guide/provisioning-the-jetbrains-offline-activation-code/

Reply:
Here, as the output destination is the user (not piped to another tool) it may not be a problem.